FourthTask/figures/types/types.go

```diff
@@ -0,0 +1,70 @@
+//	import $GOPATH/src/github.com/Mishamba/EPAM_GOLANG_COURSE/FourthTask/figures/types
+//	Package contains structures descriptions and constructors, interface definition
+
+package types
+
+import (
+	"errors"
+	"math"
+)
+
+const (
+	circlePanic   string = " circle can't has negative radius"
+	squarePanic   string = " square can't has negative side"
+	overflowError string = " result will be bigger, that we can coutn. sorry..."
+)
+
+type Figure interface {
+	area() (float64, error)
+	perimetr() (float64, error)
+}
+
+type Circle struct {
+	radius int
+}
+
+func CircleConstruct(newRadius int) (Circle, error) {
+	if newRadius < 0 {
+		return Circle{0}, errors.New(circlePanic)
```

Comment:
```suggestion
		return nil, ErrCircleNegative
```
`nil` can be returned on interfaces and err is already defined, if you'll choose to implement this specific suggestion.

Reply:
i'll correct it, but i don't understand. why it works? `nil` has no `Area` or `Perimetr` functions. or we will return interface with type pointer equal `nil` ?

Reply:
Duh. I don't know where is my answer here.
in short: just ingore it please, I've misread task's description.